### PR TITLE
[Testing][WIP] accept --using-mpi command-line-argument

### LINKED ISSUE #3362: Enable --using-mpi flag in the test runner
Related with #3340. 

Just for me to remember to implement it

### PR DESCRIPTION
**Pull request by philbucher**
I am refactoring the mpi-tests in the MappingApp and it would be nice to have this, eventually we need it anyway so I thought I could just do it myself :)

implements #3362 / closes #3362 

this only makes the tests accept `--using-mpi` as cmd-line argument
Also it automatically switches to the mpi-suite (e.g. `small` => `mpi_small`)

TODO for `run_tests.py`: If we run mpi then we should run the MPI-tests (e.g. `test_MappingApplication_mpi.py`), this I think should be discussed, since it also requires the specification of an mpi-cmd (`mpiexec` or `mpirun`)

Also it is not clear still with how many core the test should be launched

### CODE REVIEW

**Comment by philbucher** [comment]
also related to #4018

**Comment by philbucher** [comment]
Ping

File: kratos/python_scripts/run_tests.py
Comment (jcotela) [inline kratos/python_scripts/run_tests.py:201]: do we need the `--` in front here?

File: kratos/python_scripts/KratosUnittest.py
Comment (jcotela) [inline kratos/python_scripts/KratosUnittest.py:96]: The original idea was to place the mpi tests on separate queues, what would be your plan in this sense? (or, to put it in a different way, should mpi runs also call the serial tests of the corresponding suite?)

**Review by jcotela — COMMENTED** [review]
I am OK with the changes, but please give me your opinion on the questions in the review comments.

(sorry for accidentally closing the PR earlier, it was a misclick)

File: kratos/python_scripts/run_tests.py
Comment (philbucher) [reply kratos/python_scripts/run_tests.py:201]: hm I don't think so, but I have to check again, it has been some time since I did this

File: kratos/python_scripts/KratosUnittest.py
Comment (philbucher) [reply kratos/python_scripts/KratosUnittest.py:96]: I haven't removed them, only changed how they are selected:
=> If I run the tests with `--using-mpi` then automatically the corresponding mpi test queue is selected (see [here](https://github.com/KratosMultiphysics/Kratos/blob/core/mpi-testing-wip/kratos/python_scripts/KratosUnittest.py#L134-L135))
i.e. if you launch with `--using-mpi` then `small` gets converted to `mpi_small`

We can discuss it in person, as I said it has been some time since I did this

**Comment by philbucher** [comment]
the suites are still there: https://github.com/KratosMultiphysics/Kratos/blob/core/mpi-testing-wip/kratos/python_scripts/KratosUnittest.py#L146-L155

File: kratos/python_scripts/run_tests.py
Comment (philbucher) [reply kratos/python_scripts/run_tests.py:201]: I just checked again with the new tests in the MappingApp, it works nicely as is

**Comment by philbucher** [comment]
I would say this is ready to be merged, at least for me it works nicely

File: kratos/python_scripts/KratosUnittest.py
Comment (roigcarlo) [inline kratos/python_scripts/KratosUnittest.py]: Could you move the import to the top of the script?

File: kratos/python_scripts/run_tests.py
Comment (roigcarlo) [inline kratos/python_scripts/run_tests.py]: same

**Review by roigcarlo — COMMENTED** [review]
Just a couple of comments, other than that ok for me too

File: kratos/python_scripts/KratosUnittest.py
Comment (philbucher) [reply kratos/python_scripts/KratosUnittest.py]: done
For the record: I sometimes import things not at the top if they are not always used, which was the case her

I don't have a strong opinion on this though

File: kratos/python_scripts/run_tests.py
Comment (philbucher) [reply kratos/python_scripts/run_tests.py]: done

**Review by roigcarlo — APPROVED** [review]
I suppose its ok, if you want to wait for @jcotela approval, but looks ok to me.

**Comment by jcotela** [comment]
I am OK with the changes. My only concern is that, when running the tests with the mpi option, the serial suites are no longer run. Maybe it would make sense to always run the serial tests?

**Comment by RiccardoRossi** [comment]
shouldn't this be merged??

**Comment by philbucher** [comment]
Yes, I will have a small discussion with Jordi later then I think this is good to go

**Review by jcotela — APPROVED** [review]
:+1: I think this is good to go

**Pull request merged**
